avm/res/insights/scheduled-query-rule/main.bicep

```
@@ -55,26 +56,47 @@ param scopes array
 param severity int = 3
 
 @description('Optional. How often the scheduled query rule is evaluated represented in ISO 8601 duration format. Relevant and required only for rules of the kind LogAlert.')
-param evaluationFrequency string = ''
+param evaluationFrequency string?
 
 @description('Conditional. The period of time (in ISO 8601 duration format) on which the Alert query will be executed (bin size). Required if the kind is set to \'LogAlert\', otherwise not relevant.')
-param windowSize string = ''
+param windowSize string?
 
 @description('Optional. Actions to invoke when the alert fires.')
-param actions array = []
+param actions actionsType?
 
 @description('Required. The rule criteria that defines the conditions of the scheduled query rule.')
 param criterias object
 
-@description('Optional. Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired. If set, autoMitigate must be disabled.Relevant only for rules of the kind LogAlert.')
-param suppressForMinutes string = ''
+@description('Optional. Mute actions for the chosen period of time (in ISO 8601 duration format) after the alert is fired. If set, autoMitigate must be disabled. Relevant only for rules of the kind LogAlert.')
+param suppressForMinutes string?
 
 @description('Optional. Tags of the resource.')
 param tags object?
 
 @sys.description('Optional. Enable/Disable usage telemetry for module.')
 param enableTelemetry bool = true
 
+import { managedIdentityAllType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
+@description('Optional. The managed identity definition for this resource. You can only configure either a system-assigned or user-assigned identities, not both.')
+param managedIdentities managedIdentityAllType?
+
+var formattedUserAssignedIdentities = reduce(
+  map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }),
+  {},
+  (cur, next) => union(cur, next)
+)
+
+var identity = !empty(managedIdentities)
+  ? {
+      type: (managedIdentities.?systemAssigned ?? false)
+        ? (!empty(managedIdentities.?userAssignedResourceIds)
+            ? 'UserAssigned' // fail('You can only configure either a system-assigned or user-assigned identities, not both.') // Not supported by PSRule: https://github.com/microsoft/PSRule/issues/2840
```

Should we keep this comment around? Not sure if relevant for template consumers. I'd suggest an issue as we will likely update all modules supporting msi once PSRule bugfix is released
```suggestion
            ? 'UserAssigned'
```

---

This is intended and was already fixed in PSRule. So the code snippet would be coded back in as soon as the PSRule release goes out. Would hence like to keep it

---

I figured that was the solution to be uncommented once the fix is released. A random unrelated comment wasn't my first guess :smile: 

I suggested an issue instead since I don't see the value for the comment. I'm sure you will remember about it and fix it right away as soon as applicable, but I cannot really prefer a comment over an issue since the latter is definitely easier to track and not be forgotten. Anyway, I'll not block the PR on this so feel free to comment if you prefer otherwise.

---

Have it on my shortlist :) Trust me 😨